djot_parser/djot_html.go

```diff
@@ -91,7 +91,25 @@ var DefaultConversionRegistry = map[DjotNode]Conversion{
 	DefinitionListNode: func(s ConversionState, n func(c Children)) { s.BlockNodeConverter("dl", n) },
 	UnorderedListNode:  func(s ConversionState, n func(c Children)) { s.BlockNodeConverter("ul", n) },
 	OrderedListNode:    func(s ConversionState, n func(c Children)) { s.BlockNodeConverter("ol", n) },
-	ListItemNode:       func(s ConversionState, n func(c Children)) { s.BlockNodeConverter("li", n) },
+	ListItemNode: func(s ConversionState, n func(c Children)) {
+		class := s.Node.Attributes.Get(djot_tokenizer.DjotAttributeClassKey)
+		if class == CheckedTaskItemClass || class == UncheckedTaskItemClass {
+			s.Writer.InTag("li")(func() {
+				s.Writer.WriteString("\n")
+				s.Writer.WriteString("<input disabled=\"\" type=\"checkbox\"")
+				if class == CheckedTaskItemClass {
+					s.Writer.WriteString(" checked=\"\"")
+				}
+				s.Writer.WriteString("/>").WriteString("\n")
+				if len(s.Node.Children) > 1 {
```

Comment:
It seems to me that you should just print all children here and do not trim any because there can be more content in the subtree (I think you can get extra new line because of this - but you can just do not emit it manually and let `n(s.Node.Children)` to handle that automatically)